judge/models/problem.py

```diff
@@ -434,6 +447,32 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def is_solved_by(self, user):
+        # Return true if a full AC submission to the problem from the user exists.
+        return self.submission_set.filter(user=user.profile, result='AC', points__gte=F('problem__points')).exists()
```

Comment:
This is going to have issues with 9999/10000 in a 5 point problem or something, but I couldn't be bothered to care.